pkg/networkserver/grpc_asns.go

```diff
@@ -182,19 +213,19 @@ func (ns *NetworkServer) DownlinkQueuePush(ctx context.Context, req *ttnpb.Downl
 
 	dev, err := ns.devices.SetByID(ctx, req.EndDeviceIdentifiers.ApplicationIdentifiers, req.EndDeviceIdentifiers.DeviceID,
 		[]string{
+			"mac_state",
+			"pending_mac_state",
+			"pending_session",
 			"queued_application_downlinks",
-			"mac_state.device_class",
```

Comment:
If stored `mac_state` is `nil`, `dev.MACState` will be equal to `&ttnpb.MACState{}`, which causes panic in validation.